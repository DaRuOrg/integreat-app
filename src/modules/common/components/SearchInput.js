import React from 'react'
import PropTypes from 'prop-types'
import FontAwesome from 'react-fontawesome'
import { translate } from 'react-i18next'

import style from './SearchInput.css'

class SearchInput extends React.Component {
  static propTypes = {
    filterText: PropTypes.string.isRequired,
    onFilterTextChange: PropTypes.func.isRequired,
    spaceSearch: PropTypes.bool
  }

  render () {
    const {t} = this.props
    return (
      <div className={this.props.spaceSearch ? style.searchSpacing : undefined}>
        <div className={style.search}>
          <FontAwesome className={style.searchIcon} name='search' />
          <input type='text' placeholder={t('search')} className={style.searchInput} defaultValue={this.props.filterText}
                 onChange={(event) => this.props.onFilterTextChange(event.target.value)} />
        </div>
      </div>
    )
  }
}

export default translate('common')(SearchInput)
