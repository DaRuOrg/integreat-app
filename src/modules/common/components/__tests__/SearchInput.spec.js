import React from 'react'
import { shallow } from 'enzyme'
import SearchInput from '../SearchInput'

describe('SearchInput', () => {
  test('should render', () => {
    expect(shallow(<SearchInput filterText={'Test'} onFilterTextChange={() => {}} />)).toMatchSnapshot()
  })

  test('should render and space search', () => {
    expect(shallow(<SearchInput filterText={'Test'}
                                onFilterTextChange={() => {}}
                                spaceSearch />)).toMatchSnapshot()
  })
})
