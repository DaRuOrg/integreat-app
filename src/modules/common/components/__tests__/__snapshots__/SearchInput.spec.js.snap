// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchInput should render 1`] = `
<SearchInput
  filterText="Test"
  onFilterTextChange={[Function]}
  t={[Function]}
/>
`;

exports[`SearchInput should render and space search 1`] = `
<SearchInput
  filterText="Test"
  onFilterTextChange={[Function]}
  spaceSearch={true}
  t={[Function]}
/>
`;
